meeko/linked_rdkit_chorizo.py

```diff
@@ -343,6 +344,88 @@ def rectify_charges(q_list, net_charge=None, decimals=3) -> list[float]:
     return charges_dec
 
 
+def get_updated_positions(residue, new_positions: dict): 
```

Comment:
(no rush to reply, just comment for record) I really like this. It works for dihedral rotor types of conformational changes. 
Is it safe to remove the `max one n2 absent from new_positions` restriction in condition A? If we're not restricting the number of n2, then this will also work for ring puckering.

Reply:
I really like `get_updated_positions`. I have following comments: 
1. Dropping `max one n2 absent from new_positions` might enable ring puckering. I don't 100% understand why the restriction was set for the condition, so maybe it's not safe to do so
2. Enumerating atoms in new_positions (not all atoms) and add hydrogens to set h_to_update might be slightly more efficient. 
3. n2 might not be heavy atom. I think hydrogens should be counted. When n2 is a hydrogen and a new position is being assigned, then the hydrogen needs to go to h_to_update.

Reply:
Excellent insight here, thanks.
1. Great point, I agree. I only added the restriction because I saw the hydrogen on a C-alpha get updated and I wished it kept the original position. But a better way to prevent the H on C-alpha from being updated is probably to remove the C-alpha from the `new_positions` argument, delegating the decision to the caller of `get_updated_positions()`, which feels right because that is where sidechain knowledge exists, and that sidechain knowledge is that C-alphas don't move because they are in the ROOT block of the PDBQT.

3. Agree.

Will implement 1. and 3. I didn't understand what you meant with 2., could you rephrase?